backend/peephole/peephole_rules.ml

```diff
@@ -196,6 +196,42 @@ let fold_intop_imm (cell : Cfg.basic Cfg.instruction DLL.cell) =
     else None
   | _ -> None
 
+let remove_intop_neutral_element (cell : Cfg.basic Cfg.instruction DLL.cell) =
+  (* CR-someday xclerc for xclerc: it is not clear we want these rewrites to
+     happen here. Indeed, it is probably better to avoid these useless
+     operations when generating CMM, but this is currently blocked on an
+     upcoming refactoring there and it is simple enough to do here. *)
+  match U.get_cells cell 1 with
+  | [cell] -> (
+    let instr = DLL.value cell in
+    match instr.desc with
+    | Op (Intop_imm (op, imm))
+      when Array.length instr.arg = 1
+           && Array.length instr.res = 1
+           && U.are_equal_regs
```

Comment:
We should update `U.are_equal_regs` to `same_loc_fatal_on_unknown`

Reply:
Indeed, done.